### ISSUE TITLE
Think of interesting markets

### ISSUE THREAD

**Issue by kalekundert** [kxgames/seacow_spreadsheet_prototype#19]
We still don't have a great understanding of how to tweak a market's parameters (i.e. global supply, global demand, elasticity) to get interesting behavior.  Here, we can brainstorm ideas for different ways to set these values, and what effects they'd be expected to have.  The goal is to end up with a list of ideas to try in-game, to find those that are strategically interesting.

**Comment by alexmitchell** [comment]
Three variables to consider for each market:
1) World supply -- Heavily influences a player's share in a market
2) World demand -- Heavily influences the total payout.
3) Market interaction (aka. the elasticity between markets)

Volatile markets are characterized by:
- Low world supply with high world demand?

Stable markets are characterized by:
- High world supply with low world demand?


- Not sure how elasticity would factor in to these two...

**Comment by alexmitchell** [comment]
A few semi-disjointed thoughts on markets in Seacow that would hopefully make them more interesting:

- There could be more markets than what would be feasible for players to enter. The cost of entering an industry is high enough that it is not possible to enter all the markets early in the game (as I have tried to do in all our previous play testing matches haha). For example, a player might be in 3-5 markets out of 10-15 available by mid-game in the prototype. (BUT I suppose a caveat is that there can't be so many that the player can't understand the markets... Which is a pretty low threshold now that I think about it.)

- Demand from Game Indices:
  - Demand for Stable Markets could be influenced by slow-moving game variables like population, cumulative infrastructure, territory extent, and/or some sort of smoothed research activity index. Demand can come from the world or from individual players, depending on the variable?

  - Demand for Volatile Markets could be influence by game variables that can change rapidly like a war/peace index, the rate of change of other industries (i.e. projection/speculation), rolling average of new infrastructure projects (rate of building creation), and/or whims of society (i.e. smoothed random demand). Demand can come from the world or from individual players, depending on the variable?

  - I suppose one way to thing about it is that players directly control all aspects of their supply in any market but only have indirect and incomplete influence on demand.

- Number of interactions:
  - Stable Markets have few if any direct interactions (dependencies) with other markets. By definition of a stable market, any incoming interactions would be inelastic. Therefore, it wouldn't really do anything beyond making the market network graph more confusing to look at and there isn't much point in creating the dependency.
 
  - Volatile Markets directly interact with a few other markets. Interactions with Stable Markets are positive but intermediate strength. Interactions with other Volatile Markets can be strongly positive or negative and are not necessarily reciprocated. We could split it up so the total magnitude of positive or negative interactions are the same, but the number and strength of interactions vary. e.g. one strong negative vs two intermediate negatives

- (Half-baked) Maybe there are only a handful of markets overall instead of dozens, but there are no (or few) interactions between them at the start of the game. Players then unlock connections through research. The connections are the same for all players and one player unlocking a connection unlocks it for everyone. A player can only unlock a connection if they are in one of the two markets involved. Although I'm not convinced it entirely makes sense in terms of accuracy, it would prevent overwhelming players with a dense mesh of markets and would provide another way for players to differentiate and/or mess with eachother. (For example, it would be like growing a uranium market doesn't impact the oil market until a player researches nuclear energy)
  - Alternatively, when a market is unlocked (through "initial investment" aka research), certain connections are unlocked with it depending on the specific research completed. For example, unlocking the uranium industry through weapons research would provide different connections than energy research (and researching both unlocks both sets of connections). Casually assuming there can be multiple pathways to unlock markets of course. Which is an interesting idea imo...



PS: I have a nagging feeling that I'm not using the markets correctly....

**Comment by alexmitchell** [comment]
Also I should add that I tried making a 14 market graph (6 stable A-F, 8 volatile S-Z). It turned out chaotic and difficult to understand.

![market_graph_no_clusters](https://user-images.githubusercontent.com/530950/232353880-edeb5e80-a87f-41bd-825c-ab5891044632.png)

**Comment by alexmitchell** [comment]
I don't know about you, but I'm finding it difficult to design a system when everything is so abstracted. Any set of balanced interactions is valid, so which one to choose? How to define balanced? How to avoid a rat's nest of edges in the graph? Hence, randomized choices.

But we don't want randomized markets in the final game, correct? It might be a good idea to nail down a list of markets now before figuring out the connections or balancing.

We have attempted to generate a list of industries before. Assuming I'm remembering prior attempts correctly, I think part of the reason that they didn't stick was because of the "scale" of the markets weren't really considered. As in Rice, Wheat, and Cattle might be huge interacting industries on their own in the real world, but they are sourced in a similar way (farming) and essentially do the same thing (feed people). Same with Wool vs Cotton. But those industries operate with a much narrower scope than vague industries like "Manufacturing" or "Construction" and have absolutely nothing to do with industries like "Computers" or "Diamonds". Distinguishing between similar narrow markets feels like it creates too much clutter for the added strategic value, especially when trying to focus on industries elsewhere in the market graph. Additionally, mixing scales can create confusion when trying to design the interactions.

Perhaps one solution is to keep all these narrow industries but divide them into a handful of sectors. The sectors as a whole interact positively following a general trend of more processed sectors generate demand in less processed sectors (inverse of typical volatility trend too). The influence going into a sector is applied evenly to all markets within in that sector. Influence coming out of a sector is the sum of all the markets in that sector. However, market interactions within one sector can be complex with many positives and negatives at variable strengths. When viewing the markets, the sectors have there relatively simple connections plotted and each sector can be expanded to show the subgraph representing all the markets in that sector. There are no direct edges shown between two markets in different sectors. It would look similar to a disjointed graph.

Some example names for sectors and markets:
- Agriculture (Wheat, Rice, Cotton, Wool, Cattle, Hog, Poultry, Fish, etc.) -- Generally stable
- Natural Resources (Wood, Stone, Aggregate, Petroleum, Water?, etc.) -- Generally stable
- Strategic (Uranium, Iron, Copper, Cobalt, Essential Minerals? etc.) -- Generally stable?
- Manufacturing (Textiles, Machinery, Vehicles, Plastics, etc.) -- Intermediate volatility
- Infrastructure (Construction, Transportation, Energy, Space, etc.) -- Intermediate volatility
- Technology (Electronics, Robotics, Computers, etc.) -- Intermediate volatility
- Luxury (Precious Gems, Precious Metals, Jewelry, Alcohol, Dies, Spices, etc.) -- Volatile


I think this might be a decent compromise between having complex interactions between some markets and overall playability of the market system.

To circle back to where this post started, organizing the industries into sectors might also help us with designing the market interactions because now we aren't mixing industries which operate at very different scales.

**Comment by kalekundert** [comment]
I think this is a reasonable idea; basically make the system easier to understand by adding a level of hierarchy.

I go back and forth on the idea of using real markets like sugar/cotton/oil and abstract markets like A/B/C, although I think I still prefer the abstract ones.  My thinking is that our main priority right now is to find some sets of market interactions that lead to interesting game play.  Specifically, I'm imagining that we'll find a handful of patterns, each involving just 2-4 markets, then we'll combine those to get a full set of interactions for ≈20 markets.  I think that abstract markets will make this process easier because we won't have to worry about the interactions making sense, we can just say "A reduces demand for B but increases demand for C", etc.  Once we have interactions that are fun, we can think of names that best match how they actually work (and try to simplify things as much as possible).

And yeah, the random markets are definitely not how we want to do the final game.  I think that was just a stopgap to have something to play, but we definitely need to design the market interactions by hand.

**Comment by alexmitchell** [comment]
Now that you say it, I'm pretty sure you told me that exact thing during a call about developing small patterns that can be assembled into a larger system -- and then I forgot after a few days. Oops! Anyway, you're right that it would be better to find interesting patterns first. I'm just having trouble thinking of patterns that are small AND balanced AND interesting. Maybe we need to relax the balanced requirement a little. Things like:
- It's okay for a small pattern to give one or two markets an advantage or disadvantage, but then we can balance those out when assembling the small patterns into a larger system.
- Balancing can come through another mechanism like side effects not directly related to market performance. Though it would be hard to do at this stage of the prototype. For example, a small pattern gives the construction industry a disadvantage, but investing in that industry improves build speeds. Or maybe for the prototype, things like: industry A has an advantage from the patterns, but it also increases soldier costs; industry B has a disadvantage but increases building efficiency (decreased resource requirement per building)

**Comment by kalekundert** [comment]
Yeah, I think it'll be important to balance markets based on factors besides pure ROI.  Some more ideas:

- Map scarcity: Better markets are harder to find the raw resources for, which rewards players for scouting/expanding.  This actually is something that our prototype could handle pretty well.
- Required for military production: Maybe you need a supply of steel/oil/etc. to build military units, but those markets don't perfrom well.
